src/map/entities/charentity.cpp

```diff
@@ -462,6 +462,32 @@ void CCharEntity::resetPetZoningInfo()
     petZoningInfo.jugSpawnTime = 0;
     petZoningInfo.jugDuration  = 0;
 }
+
+bool CCharEntity::shouldPetPersistThroughZoning()
+{
+    PET_TYPE petType;
+    auto     PPetEntity = dynamic_cast<CPetEntity*>(PPet);
+
+    if (PPetEntity == nullptr && !petZoningInfo.respawnPet)
+    {
+        return false;
+    }
+
+    if (PPetEntity != nullptr)
+    {
+        petType = PPetEntity->getPetType();
+    }
+    else // petZoningInfo.respawnPet == true
+    {
+        petType = petZoningInfo.petType;
+    }
+
+    return petType == PET_TYPE::WYVERN ||
+           petType == PET_TYPE::AVATAR ||
+           petType == PET_TYPE::AUTOMATON ||
+           (petType == PET_TYPE::JUG_PET && settings::get<bool>("map.KEEP_JUGPET_THROUGH_ZONING"));
```

Comment:
Thanks for adding the check. Should we add the `KEEP_JUGPET_THROUGH_ZONING` default value to settings/default/map.lua?

Reply:
I already added one a while ago. [settings/default/map.lua#L254](https://github.com/LandSandBoat/server/blob/base/settings/default/map.lua#L254)

Reply:
Ah nice. I didn't know this or I would have used it. Thanks.